### PR TITLE
[feature](auth) support user password policy and alter user stmt

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1. Support password policy 

    Support setting password policy for user.

	* PASSWORD_HISTORY: Whether to allow user history password.
	* PASSWORD_EXPIRE: Set password expiration.
	* FAILED_LOGIN_ATTEMPTS and PASSWORD_LOCK_TIME: Lock the account after login failed for specific times.

2. Support `ALTER USER` stmt

	Now, you can use `ALTER USER` stmt to:
	
	* set password
	* set role
	* set password policy
	* unlock account

More details, please see documents.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
3. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [x] Yes (If Yes, please explain WHY): FE meta version changed.
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/PasswordOptions.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    public int getAccountUnlocked() {
+        return accountUnlocked;
+    }
+
+    public void analyze() throws AnalysisException {
+        if (expirePolicySecond < -2L) {
+            throw new AnalysisException("The password expire time must be DAFAULT or >= 0");
+        }
+        if (historyPolicy < -2 || historyPolicy > HistoryPolicy.MAX_HISTORY_SIZE) {
+            throw new AnalysisException("The password history number must be DEFAULT or between 0 and 10");
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/PasswordOptions.java:109]:
```suggestion
            throw new AnalysisException("The password history number must be DEFAULT or between 0 and " + HistoryPolicy.MAX_HISTORY_SIZE);
```

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 2]
```diff
@@ -1650,6 +1670,145 @@ opt_user_role ::=
...
+    opt_password_lock_time:passwdLockTime
+    opt_lock_account:lockAccount
+    {:
+        RESULT = new PasswordOptions(expirePolicy, historyPolicy, reusePolicy, loginAttempts, passwdLockTime, lockAccount);
+    :}
+    ;
+
+opt_passwd_expire_policy ::=
+    {:
+        RESULT = -2L;
```
Comment (AshinGau) [inline fe/fe-core/src/main/cup/sql_parser.cup:1687]: Does `cup` has constant variable to identify `-2`? Or import a static value from embedded java code.

Comment (morningman) [reply fe/fe-core/src/main/cup/sql_parser.cup:1687]: Done, use PasswordOptions.UNSET

File: docs/zh-CN/docs/advanced/variables.md [hunk 3]
```diff
@@ -520,4 +523,27 @@ SELECT /*+ SET_VAR(query_timeout = 1, enable_partition_cache=true) */ sleep(3);
...
+
+* `default_password_lifetime`
+
+ 	默认的密码过期时间。默认值为 0，即表示不过期。单位为天。该参数只有当用户的密码过期属性为 DEFAULT 值时，才启用。如：
+ 	
+ 	```
+ 	CREATE USER user1 IDENTIFIED BY "12345" PASSWORD_EXPIRE DEFAULT;
+ 	ALTER USER user1 PASSWORD_EXPIRE DEFAULT;
+	```
+* `password_history`
```
[the remark this answers is not in the document]
Comment (morningman) [reply docs/zh-CN/docs/advanced/variables.md:537]: This is same as mysql
https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_password_history

**Review by stalary — DISMISSED** [review]
LGTM

**Review by stalary — APPROVED** [review]

**Pull request merged**